### PR TITLE
Uhm 3031 complex obs backup

### PR DESCRIPTION
**Pull request by jmbabazi**

### CODE REVIEW

**Review by mogoodrich — APPROVED** [review]
@jmbabazi  added some comments but generally looks good to me.. and I'm definitely supportive of adding more logging and mailing... we can monitor after we deploy and see how it works.

Also, I should remember how we have this set up, but my assumption is that this just makes the local backup, and there's something else (azcopy?) that just copies the entire backups directory to the cloud?

File: mirebalais-modules/openmrs/templates/mysqlbackup.sh.erb [hunk 1]
```diff
@@ -23,27 +27,75 @@ fi
...
+        exit
+else
+        # link the current to the latest dump
+        rm -f ${DIR}/current/${CURRENT_FILENAME}
+        ln -s ${DIR}/sequences/${FILENAME} ${DIR}/current/${CURRENT_FILENAME}
 fi
 
+## backup complex obs directory
+#####################################
+COMPLEX_OBS_DIR='/home/tomcat7/.OpenMRS/complex_obs'
```
[the remark this answers is not in the document]
Comment (jmbabazi) [reply mirebalais-modules/openmrs/templates/mysqlbackup.sh.erb]: Thanks @mseaton , we use /home/<%= @tomcat %>, changed to that

File: mirebalais-modules/openmrs/templates/mysqlbackup.sh.erb [hunk 2]
```diff
@@ -23,27 +27,75 @@ fi
...
+COMPLEX_OBS_DIR1='${DIR}/sequences/complex_obs'
```
[the remark this answers is not in the document]
Comment (jmbabazi) [reply mirebalais-modules/openmrs/templates/mysqlbackup.sh.erb]: Done!

File: mirebalais-modules/openmrs/templates/mysqlbackup.sh.erb [hunk 3]
```diff
@@ -23,27 +27,75 @@ fi
...
-# Copy backup to remote location
-# (This is generally handled by crashplan in our newer implementations and not used)
-if [ "$REMOTE_DB_SERVER" != "foo" ]
-then
-	scp ${DIR}/current/${CURRENT_FILENAME} ${REMOTE_DB}/${FILENAME}
-fi
\ No newline at end of file
+# remove this dir so that it is created next time
+if [ $? -eq 0 ]; then
+    rm -rf ${COMPLEX_OBS_DIR1}
```
[the remark this answers is not in the document]
Comment (jmbabazi) [reply mirebalais-modules/openmrs/templates/mysqlbackup.sh.erb]: Made the changes @mogoodrich https://github.com/PIH/mirebalais-puppet/pull/196/commits/c991e310b589d2b42cb430363d80660b23581a2e

**Pull request merged**